ui_test/src/comments.rs

```diff
@@ -0,0 +1,111 @@
+    /// Only run this test if all of these filters apply
+    pub only: Vec<String>,
+    /// Generate one .stderr file per bit width, by prepending with `.64bit` and similar
+    pub stderr_per_bitwidth: bool,
+    /// Additional flags to pass to the executable
+    pub compile_flags: Vec<String>,
+    /// Additional env vars to set for the executable
+    pub env_vars: Vec<(String, String)>,
+    /// Normalizations to apply to the stderr output before emitting it to disk
+    pub normalize_stderr: Vec<(Regex, String)>,
+    /// An arbitrary pattern to look for in the stderr.
+    pub error_pattern: Option<(String, usize)>,
+    pub error_matches: Vec<ErrorMatch>,
+}
+
+impl Comments {
+    pub fn parse(path: &Path) -> Self {
+        let mut this = Self::default();
+        let content = std::fs::read_to_string(path).unwrap();
+        let error_pattern_regex =
+            Regex::new(r"//(\[(?P<revision>[^\]]+)\])?~[|^]*\s*(ERROR|HELP|WARN)?:?(?P<text>.*)")
+                .unwrap();
+        for (l, line) in content.lines().enumerate() {
+            if let Some(revisions) = line.strip_prefix("// revisions:") {
+                assert_eq!(
+                    this.revisions,
+                    None,
+                    "{}:{l}, cannot specifiy revisions twice",
+                    path.display()
+                );
+                this.revisions =
+                    Some(revisions.trim().split_whitespace().map(|s| s.to_string()).collect());
+            }
+            if let Some(s) = line.strip_prefix("// ignore-") {
+                let s = s
+                    .split_once(|c: char| c == ':' || c.is_whitespace())
+                    .map(|(s, _)| s)
+                    .unwrap_or(s);
+                this.ignore.push(s.to_owned());
+            }
+            if let Some(s) = line.strip_prefix("// only-") {
+                let s = s
+                    .split_once(|c: char| c == ':' || c.is_whitespace())
+                    .map(|(s, _)| s)
+                    .unwrap_or(s);
+                this.only.push(s.to_owned());
+            }
+            if line.starts_with("// stderr-per-bitwidth") {
+                assert!(
+                    !this.stderr_per_bitwidth,
+                    "{}:{l}, cannot specifiy stderr-per-bitwidth twice",
+                    path.display()
+                );
+                this.stderr_per_bitwidth = true;
+            }
+            if let Some(s) = line.strip_prefix("// compile-flags:") {
+                this.compile_flags.extend(s.split_whitespace().map(|s| s.to_string()));
+            }
+            if let Some(s) = line.strip_prefix("// rustc-env:") {
+                for env in s.split_whitespace() {
+                    if let Some((k, v)) = env.split_once('=') {
+                        this.env_vars.push((k.to_string(), v.to_string()));
+                    }
+                }
+            }
+            if let Some(s) = line.strip_prefix("// normalize-stderr-test") {
```

Comment:
Why does this not have a ":" but all the others do?
In rustc it can optionally have a colon. If we make it mandatory for all other options here, IMO we should do that for this option as well.

Reply:
"backcompat"... I just tried to replicate what the test suite does and look into consistency later.